include/xtensor/xsort.hpp

```diff
@@ -423,7 +423,7 @@ namespace xt
             std::copy(e.shape().cbegin() + std::ptrdiff_t(axis) + 1, e.shape().cend(), alt_shape.begin() + std::ptrdiff_t(axis));
 
             result_type result = result_type::from_shape(std::move(alt_shape));
-            auto result_iter = result.begin();
+            auto result_iter = result.template begin<L>();
```

Comment:
cc @DavisVaughan let me know if this does what you wanted.

Reply:
This looks like what I want. The only thing I wasn't sure about is whether the `result_type` should be created with the `L` template parameter too.

Currently, the `result_type` is specified here:
https://github.com/QuantStack/xtensor/blob/cae132d888543dfe61a00cbe050f76b1c476bfda/include/xtensor/xsort.hpp#L410

And `argfunc_result_type` seems to just create `xarray<std::size_t>` which I would assume uses the default layout? 
https://github.com/QuantStack/xtensor/blob/cae132d888543dfe61a00cbe050f76b1c476bfda/include/xtensor/xsort.hpp#L370

I don't think it is going to matter for me right now, just wanted to point it out in case it should be changed.

Reply:
OK, I see what you mean. So, you are correct about the fact that the returned value has the `XTENSOR_DEFAULT_LAYOUT`.

One solution to this, which could be done in a separate PR is to optionally let the user specify the return type as a second template parameter after the traversal order, so one could directly specify a R, Python, or Julia array.

Reply:
That makes sense to me. 👍